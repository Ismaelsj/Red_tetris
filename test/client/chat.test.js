import 'jsdom-global/register';
import React from 'react'
import { shallow, mount } from 'enzyme';
import "babel-polyfill"

import Chat from '../../src/client/components/chat/chat'

import { Provider } from 'react-redux'
import io from 'socket.io-client';
import storeConfig from '../../src/client/middlewares/store'

const serverParams = {
    host: '0.0.0.0',
    port: 8881,
    url: 'http://' + '0.0.0.0' + ':' + '8881' 
}

const { url } = serverParams

const socket = io(url)

const store = storeConfig(socket)

describe('Chat', () => {
    test('chat => render: props -> ...', () => {
        const component = mount(
            <Provider store={store}>
                <Chat />
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });
});