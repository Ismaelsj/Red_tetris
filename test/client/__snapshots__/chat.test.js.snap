// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat chat => redner: props -> ... 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Chat)
    onNewMessage={null}
    onRemoveMsgNot={null}
    roomState={
      Object {
        "conversation": Array [],
        "msgNot": 0,
        "roomName": "Jean",
      }
    }
    user={
      Object {
        "socket": null,
        "userName": "Jean",
      }
    }
  >
    <Chat
      onNewMessage={[Function]}
      onRemoveMsgNot={[Function]}
      roomState={
        Object {
          "conversation": Array [],
          "isChanging": false,
          "isManager": false,
          "msgNot": 0,
          "players": null,
          "roomName": null,
          "userName": "",
        }
      }
      user={
        Object {
          "isAuth": false,
          "room": "",
          "socket": null,
          "userName": "",
        }
      }
    >
      <div>
        <button
          className="open-button"
          onClick={[Function]}
          type="button"
        >
          Chat
        </button>
        <div
          className="chat-popup"
          id="myForm"
        >
          <div
            className="form-container"
          >
            <div
              className="messageBox"
            >
              <div
                className="messages"
              >
                <ul />
              </div>
              <div
                className="input"
              >
                <Divider
                  horizontal={true}
                >
                  <div
                    className="ui horizontal divider"
                  >
                    <Icon
                      as="i"
                      inverted={true}
                      name="chat"
                    >
                      <i
                        aria-hidden="true"
                        className="chat inverted icon"
                        onClick={[Function]}
                      />
                    </Icon>
                  </div>
                </Divider>
                <form
                  className="inputMsg"
                  onSubmit={[Function]}
                >
                  <Input
                    fluid={true}
                    inverted={true}
                    onChange={[Function]}
                    placeholder="Message..."
                    type="text"
                  >
                    <div
                      className="ui fluid inverted input"
                    >
                      <input
                        onChange={[Function]}
                        placeholder="Message..."
                        type="text"
                      />
                    </div>
                  </Input>
                </form>
              </div>
            </div>
            <button
              className="btn cancel"
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </Chat>
  </Connect(Chat)>
</Provider>
`;

exports[`Chat chat => render: props -> ... 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Chat)>
    <Chat
      onNewMessage={[Function]}
      onRemoveMsgNot={[Function]}
      roomState={
        Object {
          "conversation": Array [],
          "isChanging": false,
          "isManager": false,
          "msgNot": 0,
          "players": null,
          "roomName": null,
          "userName": "",
        }
      }
      user={
        Object {
          "isAuth": false,
          "room": "",
          "socket": null,
          "userName": "",
        }
      }
    >
      <div>
        <button
          className="open-button"
          onClick={[Function]}
          type="button"
        >
          Chat
        </button>
        <div
          className="chat-popup"
          id="myForm"
        >
          <div
            className="form-container"
          >
            <div
              className="messageBox"
            >
              <div
                className="messages"
              >
                <ul />
              </div>
              <div
                className="input"
              >
                <Divider
                  horizontal={true}
                >
                  <div
                    className="ui horizontal divider"
                  >
                    <Icon
                      as="i"
                      inverted={true}
                      name="chat"
                    >
                      <i
                        aria-hidden="true"
                        className="chat inverted icon"
                        onClick={[Function]}
                      />
                    </Icon>
                  </div>
                </Divider>
                <form
                  className="inputMsg"
                  onSubmit={[Function]}
                >
                  <Input
                    fluid={true}
                    inverted={true}
                    onChange={[Function]}
                    placeholder="Message..."
                    type="text"
                  >
                    <div
                      className="ui fluid inverted input"
                    >
                      <input
                        onChange={[Function]}
                        placeholder="Message..."
                        type="text"
                      />
                    </div>
                  </Input>
                </form>
              </div>
            </div>
            <button
              className="btn cancel"
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </Chat>
  </Connect(Chat)>
</Provider>
`;
